pkg/cmd/infra/deployer/deployer.go

```diff
@@ -196,5 +203,53 @@ func (d *Deployer) Deploy(namespace, deploymentName string) error {
 	} else {
 		glog.Infof("Deploying from %s to %s (replicas: %d)", deployutil.LabelForDeployment(from), deployutil.LabelForDeployment(to), desiredReplicas)
 	}
-	return strategy.Deploy(from, to, desiredReplicas)
+	if err := strategy.Deploy(from, to, desiredReplicas); err != nil {
+		return err
+	}
+	return d.after(config, to)
```

Comment:
Yeah, the array of after hooks with Lifecycle is probably best.  It means
lifecycle hooks can tag images - I think I'm ok with that.

On Mon, Feb 1, 2016 at 10:52 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/cmd/infra/deployer/deployer.go
> <https://github.com/openshift/origin/pull/6925#discussion_r51434887>:
>
> > @@ -196,5 +203,53 @@ func (d *Deployer) Deploy(namespace, deploymentName string) error {
> >  	} else {
> >  		glog.Infof("Deploying from %s to %s (replicas: %d)", deployutil.LabelForDeployment(from), deployutil.LabelForDeployment(to), desiredReplicas)
> >  	}
> > -	return strategy.Deploy(from, to, desiredReplicas)
> > +	if err := strategy.Deploy(from, to, desiredReplicas); err != nil {
> > +		return err
> > +	}
> > +	return d.after(config, to)
>
> basically, if I want to fail a deployment because of a post hook, I should
> be allowed to.
>
> Agreed. This means hooks/after should hold an option for this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6925/files#r51434887>.
>